csharp/src/Apache.Arrow/Arrays/ArrowArrayFactory.cs

```diff
@@ -58,8 +58,9 @@ public static IArrowArray BuildArray(ArrayData data)
                     return new StructArray(data);
                 case ArrowTypeId.Union:
                     return new UnionArray(data);
-                case ArrowTypeId.Date64:
                 case ArrowTypeId.Date32:
+                    return new Date32Array(data);
+                case ArrowTypeId.Date64:
```

Comment:
Can we make the same change for `Date64Array` if it isn't too much work? (and add a unit test)

Reply:
I thought Date64Array writing was not implemented, so it wouldn't make sense to add it

Reply:
It's one more line of code to enable `Date64Array` in `ArrowStreamWriter`. I see no reason why we wouldn't want this enabled if we have `Date32Array` working.

Check out https://github.com/apache/arrow/compare/master...eerhardt:Date32And64, which also adds unit tests for reading and writing both `Date64` and `Date32` arrays.

Reply:
@eerhardt - any eta on a new nuget publish?  I would rather not have a custom local build  - I am currently just using strings instead of dates

Reply:
The nuget gets published along with the broader Arrow release activities. I believe they are working on releasing `0.15` right now. So if this change would get in soon, it _may_ be included in the `0.15` release.

If you want to copy my code above -https://github.com/apache/arrow/compare/master...eerhardt:Date32And64 - into this change, I will sign off on this PR and it will get merged. However, I can't sign off on this until it has a unit test ensuring the change is good (which my code change adds tests for).

If you don't want those changes, I can open a new PR for it myself. But no matter what we need to add tests.